node/src/proxy_server/mod.rs

```diff
@@ -2241,26 +2552,87 @@ mod tests {
                 .proxy_server(proxy_server_mock)
                 .neighborhood(neighborhood_mock)
                 .build();
-            // Get the add_route recipient so we can partially mock it...
-            let add_route_recipient = peer_actors.proxy_server.add_route;
+            // Get the dns_retry_result recipient so we can partially mock it...
+            let dns_retry_result_recipient = peer_actors.proxy_server.route_result_sub;
             peer_actors.proxy_server = ProxyServer::make_subs_from(&subject_addr);
-            peer_actors.proxy_server.add_route = add_route_recipient; //Partial mocking
+            peer_actors.proxy_server.route_result_sub = dns_retry_result_recipient; //Partial mocking
             subject_addr.try_send(BindMessage { peer_actors }).unwrap();
 
             subject_addr.try_send(msg_from_dispatcher).unwrap();
 
             system.run();
         });
 
-        let expected_add_route_message = AddRouteMessage {
+        let expected_dns_retry_result_message = RouteResultMessage {
             stream_key,
-            route: route_query_response.unwrap(),
+            result: Ok(route_query_response.unwrap()),
         };
 
         proxy_server_awaiter.await_message_count(1);
         let recording = proxy_server_recording_arc.lock().unwrap();
-        let record = recording.get_record::<AddRouteMessage>(0);
-        assert_eq!(record, &expected_add_route_message);
+        let message = recording.get_record::<RouteResultMessage>(0);
+        assert_eq!(message, &expected_dns_retry_result_message);
+    }
+
+    #[test]
+    fn proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route() {
+        let test_name = "proxy_server_sends_a_message_when_dns_retry_cannot_find_a_route";
+        let cryptde = main_cryptde();
+        let http_request = b"GET /index.html HTTP/1.1\r\nHost: nowhere.com\r\n\r\n";
+        let (proxy_server_mock, _, proxy_server_recording_arc) = make_recorder();
+        let route_query_response = None;
+        let (neighborhood_mock, _, _) = make_recorder();
+        let neighborhood_mock = neighborhood_mock
+            .stop_on_message(type_id!(RouteResultMessage))
```

Comment:
Please, also delete the method `.stop_on_message()`, it can be easily substituted by the old `.system_stop_conditions()`. We don't want to have multiple ways of how we can do the same thing.

Your new infrastructure is used just seldomly but the other one has already turned into a standard.

Reply:
I do think that this method has vast implications throughout the codebase. It was used seldomly because we only used it for the newly written tests and the old tests that might benefit from them are untouched. We can surely replace the long and hard to read code in other tests with this quite simple and easy to use function.

I do think `.stop_on_message()` function was older than the `.system_stop_conditions()` as it was the original idea which was later expanded into the `StopConditions`. As per my experience writing tests, almost each and every test that requires a StopCondition later in the life of the test, can simply use this method.

This is my humble request to keep it, as I intend to use it inside the tests that I'll write in the future. Also, it's something that we can use to make the old tests easier to read.